assert/assertions.go

```diff
@@ -2107,3 +2107,38 @@ func buildErrorChainString(err error) string {
 	}
 	return chain
 }
+
+// NoFieldIsEmpty asserts that object, which must be a struct or eventually reference to one, has no empty exported fields.
```

Comment:
What does "empty" mean? It's not in the language [spec](https://go.dev/ref/spec). From reading the code it seems to be "zero values for most types or with length of zero for some of the container types". 

This assertion's behaviour needs to be clearly documented. I can't review it as it is because I can't tell what empty is supposed to mean. I think it's supposed to mean unpopulated struct fields? If that is the case then it doesn't work, see discussion in #1601.

Reply:
I am happy to have more explicit documentation.

One option would be to provide a [godoc link](https://pkg.go.dev/github.com/fluhus/godoc-tricks#Links) to the current [assert.Empty](https://pkg.go.dev/github.com/stretchr/testify@v1.9.0/assert#Empty) function e.g. 
> NoFieldIsEmpty asserts that object, which must be a struct or eventually reference to one, has no exported field with a value that is empty (following the definition of empty used in [Empty]).

Another option is to include the same definition used in assert.Empty in the documentation for this function e.g.
> NoFieldIsEmpty asserts that object, which must be a struct or eventually reference to one, has no exported field with a value that is empty I.e. nil, "", false, 0 or either a slice or a channel with len == 0.

Do find either these clear enough?

P.S. I am aware that "empty" is not part of the go spec. My intent was to follow a definition already used by this library (that I have found useful) rather than invent something myself.

Reply:
The godoc link would be sufficient.